docs/_posts/2019-05-03-baremetal-installation-guide.md

```diff
@@ -0,0 +1,364 @@
+    curl -O https://artifacts.oicr.on.ca/artifactory/gsi-dependencies/uk/ac/ebi/fgpt/jndi-file-factory/1.0/jndi-file-factory-1.0.jar
+
+Append the following line to `$CATALINA_HOME/bin/setenv.sh` or, if you installed Tomcat through apt, `/etc/default/tomcat8`:
+
+    JAVA_OPTS="$JAVA_OPTS -Dsecurity.method=jdbc -Xmx768M"
+
+(Update the security method if you are using LDAP or Active Directory LDAP.)
+
+Create the directory `/storage/miso` and subdirectory `/storage/miso/log`.
+
+  	mkdir -p /storage/miso/log
+
+Ensure that the user that Tomcat runs as has write permission to the storage directory. For example, if the user is in the 'tomcat' group:
+
+    chgrp -R tomcat /storage/miso/
+
+Move the following configuration files from `miso-lims/miso-web/src/main/resources` into
+the `/storage/miso/` directory:
+
+| File                      | Purpose                                                    |
+|---------------------------|------------------------------------------------------------|
+| `security.properties`     | properties to set the security environment (see below).    |
+| `submission.properties`   | properties to set the submission environment.              |
+
+## Security Environment (updating `/storage/miso/security.properties`)
+MISO can use either LDAP (`ldap`), Active Directory LDAP (`ad`), or JDBC
+(`jdbc`) as an authentication mechanism. This is set by the `-Dsecurity.method`
+noted in the previous section.
+
+If you are using JDBC (aka storing usernames and passwords in the database), set the
+security method to `jdbc`.
+The default configuration should work properly.
+
+For using LDAP, set the security method to `ldap`. Additional settings are
+needed for LDAP in the `security.properties`. Talk to your LDAP administrator.
+
+To use Active Directory, a specific kind of LDAP, set the security method to
+`ad`. Three additional settings are needed for Active Directory in the
+`security.properties` file.
+
+| Property                    | Purpose                                                    |
+|-----------------------------|------------------------------------------------------------|
+|`security.ad.emailDomain`    | Domain added to username for lookup (e.g. ad.oicr.on.ca)   |
+|`security.ad.url`            | Url for Active Directory server (e.g. ldap://ad.oicr.on.ca)|
+|`security.ad.stripRolePrefix`| Prefix to be removed from group (e.g. MISO\_)               |
+
+The search for a user is done against `userPrincipalName` which takes the form of
+an email address. To login the user will type their username and to do the lookup
+their username will be added to the domain specified in the property
+`security.ad.emailDomain`.
+
+Use the `security.ad.url` property to indicate the url for the Active Directory.
+Some valid examples are: `ad.oicr.on.ca`, `ldap://ad.oicr.on.ca:389` and
+`ldaps://ad.oicr.on.ca:636`.
+
+The groups used by MISO are `ROLE_INTERNAL` for regular users, `ROLE_EXTERNAL` for
```

Comment:
We may want to remove this line about `ROLE_EXTERNAL` as there's no longer any code that uses it.